runtime/onert/core/include/backend/basic/BackendContextHelpers.h

```diff
@@ -219,25 +234,38 @@ template <typename T_BackendContext> ITensorRegistry *genTensors(T_BackendContex
 inline void initConsts(const ir::Operands &operands,
                        const util::Set<ir::OperandIndex> &external_operands,
                        ITensorRegistry *tensor_registry,
-                       const ir::OperandIndexMap<ir::OperandIndex> & /*shared_memory_operands_map*/)
+                       const ir::OperandIndexMap<ir::OperandIndex> &shared_memory_operands_map)
 {
   operands.iterate([&](const ir::OperandIndex &ind, const ir::Operand &operand) {
-    if (external_operands.contains(ind) || !operand.isConstant())
+    const bool has_const_shared_memory =
+      shared_memory_operands_map.find(ind) != std::end(shared_memory_operands_map) &&
+      operands.at(shared_memory_operands_map.at(ind)).isConstant();
+    const bool can_be_initialized_as_const = operand.isConstant() || has_const_shared_memory;
```

Comment:
Please consider a case where input of `Reshape` is a constant like from test: https://github.com/Samsung/ONE/blob/8b1af605c2cd278322ccb7fed67fe7336227c809/tests/nnfw_api/src/GenModelTests/ModelTensorMemorySharing.test.cc#L103. In general it should be probably constant folded but still it's worth to handle it.
![image](https://github.com/user-attachments/assets/f835226d-82e7-4bcc-8171-ee3b2b9f26dc)

Reply:
Theoretically we can set data to such output constant tensor (`ExternalTensor` type) during [buildTensor method](https://github.com/Samsung/ONE/blob/e2005d88c492a574a26bf105d08ea46d7d5a6625/runtime/onert/core/src/backend/basic/StaticTensorManager.cc#L90) but we have not access there to data (`std::shared_ptr<ir::Data>` type) from an constant input of such operator. It's possible but requires changes (adding `getData` method + dealing with const limitation) in `ExternalTensor` API.
I am not sure if it's really better solution. I also have no idea for a better place to handle it.

Reply:
Output tensors of operators are never constant tensor because output data must be generated by the corresponding operator. So you don't need to check whether shared tensors are constant or not.

Reply:
> Output tensors of operators are never constant tensor because output data must be generated by the corresponding operator.
In general it makes a sense but still the kernel (at least in the current state) is called for such a case. Probably it requires additional handling in some other place. Nevertheless, thank you, good point!

Reply:
I figured out the reason why calling it requires.

Reply:
I am handling here only `ExternalTensor`.

> ExternalTensor means it don't own its data buffer internally.

I also understand it in this way so additional memory lifetime management (especially extending) is not needed.

> Both are different. Some constant operands can be ExternalTensor , but not in external_operands. Did you consider 
it?

The `external_operands` are skipped [here](https://github.com/Samsung/ONE/pull/14384/files#diff-8a69035d39f33d8974854dd98794a6090c157731ac0da047957e748c3726413dR253) so as I understand memory sharing is not applied for them. In general IMO memory sharing from external operands can be a dangerous solution since some external context control their lifetime (but I can be wrong here).
For clearance I can also skip external operands in [findSharedMemoryOperandIndexes](https://github.com/Samsung/ONE/blob/1aabefe2ec4f9c3f3197113dcf2457945f0510c4/runtime/onert/backend/cpu/SharedMemoryOperands.cc#L49) processing

Reply:
> I am handling here only `ExternalTensor`.

I guess the condition `has_const_shared_memory` means that tensors in the condition cannot be `ExternalTensor` because they cannot be constant tensor.

>  I also understand it in this way so additional memory lifetime management (especially extending) is not needed.

OK, It seems so, but I mentioned `ExternalTensor` due to the line 266. I will review it at the line.

> The external_operands are skipped here so as I understand memory sharing is not applied for them. In general IMO memory sharing from external operands can be a dangerous solution since some external context control their lifetime (but I can be wrong here).
For clearance I can also skip external operands in findSharedMemoryOperandIndexes processing

I didn't catch your wording exactly. But It seems there is no lifetime issue.

Reply:
> I guess the condition has_const_shared_memory means that tensors in the condition cannot be ExternalTensor because they cannot be constant tensor.

`has_const_shared_memory` means that a given operands is constant or such operand has source constant operand.
In topic of `ExternalTensor` you are right. I've thought wrongly that each constant tensor has `ExternalTensor` type (at least for cpu backend). I double-checked it now and you are right (`IPortableTensor` type seems to be used).

Reply:
To be honest, I didn't understand the condition exactly. I won't concern myself with that condition anymore.